src/views/requestAccess.js

```diff
@@ -1,146 +1,123 @@
+  return (
+    <div className="section-row">
+      <div className="select is-rounded">
+        <select
+          defaultValue='default'
+          onChange={e => props.selectionHandler(e.target.value)}>
+          <option value='default' disabled defaultValue>Select Role</option>
+          {
+            props.selectionOptions.map((role) => 
+            <option>{role}</option>
+            )
+          }
+        </select>
+      </div>
+    </div>
+  )
 }
 
 // Section under CONTACT
-export const InfoField = ({label, info}) => {
+export const InfoField = ({label, info, changeHandler}) => {
 	const infoField = "has-text-weight-bold " + ((label === "Phone") ? "phone-field" : (label === "Email" ? "email-field" : "name-field"));
 	return (
 		<div>
 			<hr className="line" ></hr>
 			<span className="input-field"> {label}
-				<p className={infoField} contentEditable>
-			        {info}
-			    </p>
+				<input className={`contact-input ${infoField}`} type="text" defaultValue={info} onChange={(e) => changeHandler(e.target.value)}>
+			    </input>
 		    </span>
 	    </div>
 	);
 }
 
 export const RequestAccess = (props) => {
-	// const id = props.match.params.id;
+  // Get context for API auth header
+  const userContext = useContext(UserContext);
+
+  const [roleObject, setRoleObject] = useState({});
+  const [selectionOptions, setSelectionOptions] = useState([]);
+  const [currentSelection, selectionHandler] = useState("");
+  const [fName, setFirstName] = useState("");
+  const [lName, setLastName] = useState("");
+  const [emailAddress, setEmail] = useState("");
+
+  useEffect(() => {
+    console.log(process.env.REACT_APP_API_URL)
+    axios.get(`/api/roles`)
+    .then((response) => {
+      let data = JSON.parse(response.data);
+      // Get key value object of all roles
+      setRoleObject(data);
+      // Get Role names
+      let roleArray = Object.keys(data);
+      // Removes "Pending" role
+      roleArray.shift();
+      // Get array of roles to map to selection options
+      setSelectionOptions(roleArray);
+    })
+    .catch((error) => {
+      alert(error);
+      console.log(error);
+    });
+  }, []);
+
+  const grantAccess = (role, firstName, lastName, email, id) => {
+    // Get role index from roleObject
+    let roleID = roleObject[role];
+    axios.patch(`/api/user/${id}`, {
+      role: roleID,
+      firstName: firstName,
+      lastName: lastName,
+      email: email
+    }, makeAuthHeaders(userContext))
+    .then((response) => {
+      alert("User access granted!")
+    })
+    .catch((error) => {
+      alert(error);
+      console.log(error);
+    });
+  }
 
 	const {
 		firstName,
 		lastName,
-		email
-	} = props.location.state;
+    email,
+    id
+  } = props.location.state;
+  
+  useEffect(() => {
+    setFirstName(firstName);
+    setLastName(lastName);
+    setEmail(email);
+  }, []);
 
 	return (
 		<>
 			<div className="request-page">
 				<div className="page-title"> Request Access </div>
```

Comment:
The mockup says "Request for Access" for the title of the page, I'm not sure if it makes that big of a difference but it might add clarity that the admins are looking at a list of requests to _manage_ and not a page to _make_ a request of their own 🤷‍♂️